File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.cpp
class Solution {
public:
    string reverseVowels(string s) {
        int first =0, last=s.length()-1;
        while(last>=first){
            if(!isVowel(s[first])){
                first++;
            }
            else if(!isVowel(s[last])){
                last--;
            }
            else{ swap(s[first], s[last]);
            first++;
            last--;
            }
        }
        return s;
    }

    bool isVowel(char ch){
        if(ch=='a'||ch=='e'||ch=='i'||ch=='o'||ch=='u'||ch=='A'||ch=='E'||ch=='I'||ch=='O'||ch=='U'){
            return true;
        } else{ return false;}
    }
};

File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.java
class Solution {
    public String reverseVowels(String s) {
        char[] ch = s.toCharArray();
        int l =0;
        int r = s.length()-1;
        while(r>=l){
            if(!vowel(s.charAt(l))){
                l++;
            }else if(!vowel(s.charAt(r))){
                r--;
            }else{
                swap(ch, l, r);
                l++;
                r--;
            }
        }
        return new String(ch);
    }
    public boolean vowel(char c){
        if(c=='a' || c=='e' ||c=='i' ||c=='o' ||c=='u' ||c=='A' ||c=='E' ||c=='I' ||c=='O' ||c=='U'){
            return true;
        }
        return false;
    }
    public void swap(char[] ch, int a, int b){
        char temp = ch[a];
        ch[a] = ch[b];
        ch[b] = temp;
    }
}